internal/pkg/otel/Makefile

```diff
@@ -0,0 +1,6 @@
+.PHONY: all
+
+all:
+	go run go.opentelemetry.io/collector/cmd/builder@v0.107.0 --skip-compilation --skip-get-modules --config builder-config.yaml
+	@rm main.go main_others.go main_windows.go go.mod
```

Comment:
This is a bit hacky since ocb expects to build a final binary. Having a otel collector embeded into another binary seems to be quite useful, I wonder if the folks at OpenTelemetry would be willing to support a ocb mode to build a library instead of a final binary.